maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -132,6 +137,7 @@ public DefaultMaven(
         this.buildResumptionAnalyzer = buildResumptionAnalyzer;
         this.buildResumptionDataRepository = buildResumptionDataRepository;
         this.superPomProvider = superPomProvider;
+        this.projectSelector = new ProjectSelector();
```

Comment:
The reasoning behind this is that the `ProjectSelector` is likely only going to be used in two classes (`DefaultMaven` and `DefaultGraphBuilder`) and does not require any complex configuration to be created. It's relatively lightweight and I would have liked to make it package-private as well but since `DefaultMaven` is in a different package this wasn't possible. If it preferred to subject it to DI I'll happily change it.

Reply:
Not necessary. You can leave a comment in the class that if necessary, switch to DI.